src/shims/unix/linux/epoll.rs

```diff
@@ -69,17 +69,21 @@ pub struct EpollReadyEvents {
     /// Stream socket peer closed connection, or shut down writing
     /// half of connection.
     pub epollrdhup: bool,
+    /// For stream socket, this event merely indicates that the peer
+    /// closed its end of the channel.
+    pub epollhup: bool,
```

Comment:
So, there's some subtle difference between RDHUP and HUP. But I don't really understand it... and it seems we are ignoring that difference and setting the two flags together. Could that become a problem?

Reply:
From my understanding, EPOLLRDHUP is more specific than EPOLLHUP. It will be returned even when the peer only closing its writing half. Closing wriitng half will require ``shutdown``. 

For example, 
If we use ``shutdown(sv[1], SHUT_WR);`` only ``EPOLLRDHUP`` will be triggered. If we use ``shutdown(sv[1], SHUT_RDWR);`` or ``close(sv[1])``, both ``EPOLLHUP`` and ``EPOLLRDHUP`` will be triggered. 

Since we don't have ``shutdown`` for now, I think they will have the same behaviour.